magpie/api/webhooks.py

```diff
@@ -71,20 +92,20 @@ def replace_template(params, payload):
     :return: structure containing the data with the replaced template parameters
     """
     if isinstance(payload, dict):
-        return {replace_template(params, key): replace_template(params, value)
-                for key, value in payload.items()}
+        return {replace_template(params, key): replace_template(params, value) for key, value in payload.items()}
     if isinstance(payload, list):
         return [replace_template(params, value) for value in payload]
     if isinstance(payload, str):
-        for template_param in WEBHOOK_TEMPLATE_PARAMS:
-            if template_param in params:
+        for template_param in params:
+            if template_param in WEBHOOK_TEMPLATE_PARAMS:
                 payload = payload.replace("{" + template_param + "}", params[template_param])
         return payload
     # For any other type, no replacing to do
     return payload
 
 
 def send_webhook_request(webhook_config, params, update_user_status_on_error=False):
+    # type: (WebhookConfigSettings, WebhookTemplateParameters, bool) -> None
```

Comment:
If we keep both type definitions, I think it should be WebhookLoadedSettings here.